File: problems/sorting/LC274_h_index/Solution.java
package problems.sorting.LC274_h_index;

import java.util.Arrays;

public class Solution {
  public int hIndex(int[] citations) {
    Arrays.sort(citations);

    int h = citations.length;

    for (int i = 0; i < citations.length; i++) {
      if (citations[i] >= h) {
        return h;
      } else {
        if (h == 1 && citations[i] != 0) {
          return 1;
        }
        h--;
      }
    }
    return h;
  }

  public static void main(String[] args) {
    int[] citations = new int[]{3,0,6,1,5};
    Solution s = new Solution();
    System.out.println(s.hIndex(citations));
  }
}

File: problems/sorting/LC274_h_index/solution.py
class Solution:
  def h_index(self, citations: [int]) -> int:
    citations.sort()
    h = len(citations)

    for c in citations:
      if (c >= h):
        return h
      else:
        if h == 1 and c != 0:
          return 1
        h -= 1
    
    return h

def main():
  s = Solution()
  citations = [3,0,6,1,5]
  print(s.h_index(citations))

main()